hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClientThrottlingIntercept.java

```diff
@@ -170,7 +170,7 @@ public void updateMetrics(AbfsRestOperationType operationType,
         }
         break;
       case ReadFile:
-        String range = abfsHttpOperation.getConnection().getRequestProperty(HttpHeaderConfigurations.RANGE);
+        String range = abfsHttpOperation.getRequestProperty(HttpHeaderConfigurations.RANGE);
```

Comment:
what is the abfsHttpOperation is of the type AbfsRestOperation, getConnection() removal will give error

Reply:
The name of variable was wrong, have refactored it to `httpOperation`. It is of type `HttpOperation`. `getRequestProperty` is an abstract method which the AbfsHttpOperation and AbfsAHCOperation would implement. The JDK's approach class `AbfsHttpOperation` will implement it as getConnection().getRequestProperty(key)